### ISSUE TITLE
💅 Regression: Disabling one nursery rule disables all nursery rules

### ISSUE THREAD

**Issue by chrisgrieser** [biomejs/biome#2028]
### Environment information

```bash
CLI:
  Version:                      1.6.0
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v21.7.1"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "npm/10.5.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Linter:
  Recommended:                  false
  All:                          true
  Rules:                        nursery/noConsole = "off"

Workspace:
  Open Documents:               0
```

### Rule name

all nursery rules

### Playground link
using CBS, since the playground does not allow modifying the biome config on a rule-level.
https://codesandbox.io/p/devbox/suspicious-hermann-cdnhdx?workspaceId=23165e9f-6758-4f08-8321-45be43925a49

Sidenote: Biome in the CBS is outdated, the installed version there is 1.1.2. Took me a moment to realize that that was why the nursery rules didn't exist.

### Expected result

Disabling one nursery rule should not disable all nursery rules. (The following code can also be found in the CBS.)

```ts
const x = 5;
const a = x ? true : true; // violates "noUselessTernary" from the nursery

console.info(a); // violates "noConsole" from the nursery

// Running `biome lint src` will report no issue at all,
// even though only "noConsole" has been disabled in the biome.json.
```

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
I don't think this is a bug because as the schema says the `lint.rules.all` won't enable nursery rules:

![image](https://github.com/biomejs/biome/assets/10386119/56de79aa-3364-4694-9cb8-4f95bfce493d)

If you want to enable all the rules in the nursery group, you should use `lint.rules.nursery.all`. And you can disable one single rule as expected:

```json
{
  "$schema": "./node_modules/@biomejs/biome/configuration_schema.json",
  "formatter": {
    "enabled": true
  },
  "linter": {
    "rules": {
      "nursery": { "all": true, "noConsole": "off" }
    }
  }
}
```

However I agree this behavior should be properly documented here: https://biomejs.dev/reference/configuration/#linterrulesall

**Comment by chrisgrieser** [comment]
@Sec-ant Ah, thanks for the tip. That helps indeed.

Nonetheless, this is still a bug, only a different kind of bug then, since `lint.rules.all` also enables all nursery rules for me

**Comment by Sec-ant** [comment]
> Nonetheless, this is still a bug, only a different kind of bug then, since `lint.rules.all` also enables all nursery rules for me

Yes you're right, I didn't notice that.

**Comment by Sec-ant** [comment]
I'm working on this issue 🔨.
